backend/store/memory/event_store.go

```diff
@@ -0,0 +1,366 @@
+package memory
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/snappy"
+	corev2 "github.com/sensu/core/v2"
+	"github.com/sensu/sensu-go/backend/store"
+	"golang.org/x/time/rate"
+)
+
+type memorydb struct {
+	limiter *rate.Limiter
+	data    sync.Map
+}
+
+func newMemoryDB(limiter *rate.Limiter) *memorydb {
+	return &memorydb{
+		limiter: limiter,
+	}
+}
+
+type eventEntry struct {
+	// Serialized and compressed event
+	EventBytes []byte
+
+	// Dirty indicates if the entry has yet to be written out to long term storage
+	Dirty bool
+
+	// Mu is necessary in the rare instance when multiple goroutines are competing
+	// to store different versions of the same event. This is only necessary due to
+	// a design flaw in eventd, and could possibly be mitigated there.
+	Mu sync.Mutex
+}
+
+const eventOutputTruncatedBytesLabel = "sensu.io/output_truncated_bytes"
+
+type SilenceStoreI interface {
```

Comment:
I don't think this should to be exported? And I'm not clear why we can't use the interface defined in the `store` package already?

Reply:
Yeah this is a bit crufty, I can look at refactoring things.